js/cards.js

```diff
@@ -107,32 +127,65 @@ function getOwnedCards(cards_data) {
   return ownedCards;
 }
 
-function sortCards(cards, sortType) {
+function sortCards(cards, sortType, reverse=false) {
   sortType = sortType ?? "Collector Number";
-  cards.sort(sort_functions[sortType]);
+  let before = reverse ? -1 : 1
+  let after = reverse ? 1 : -1
+  cards.sort(sort_functions[sortType](before, after));
   return cards;
 }
 
-export function showCollection(cards_data, htmlLocation) {
-  let sort = localStorage.getItem("sort");
+export function showCollection(cards_data, htmlLocation, page = 1) {
+  let sort = localStorage.getItem("sort") ?? "Collector Number";
   let fullCollection = localStorage.getItem("fullCollection") === "true";
+  let cards;
+  let reverse = false;
   let ownedCount = 0;
   for (let rarity of RARITIES) {
     ownedCount += parseInt(localStorage.getItem(`count-${rarity}`) ?? "0");
   }
   COLLECTED_CARDS_NUMBER.textContent = `Collected cards: ${ownedCount}/${cards_data.length}`;
 
+  if (sort[sort.length - 1] === "-") {
+    sort = sort.slice(0, -1)
+    reverse = true;
+  }
```

Comment:
Having trouble understanding this line. If 'sort' is one of the dropdown string values like "Rarity", when would it have "-"?

Reply:
Added comment. If the sort type has a "-" at the end, it means it's in reverse order ("Rarity (Reverse)" in the dropdown has value "Rarity-", for example). It is admittedly a bit hacky, though effective. I can change it if it's _too_ hacky.

Reply:
no changes needed, thanks for the comment! I didn't pay attention to the value="Rarity-" in the HTML before